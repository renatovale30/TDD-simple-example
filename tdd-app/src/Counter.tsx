/*
 * Talkdesk Confidential
 *
 * Copyright (C) Talkdesk Inc. 2021
 *
 * The source code for this program is not published or otherwise divested
 * of its trade secrets, irrespective of what has been deposited with the
 * U.S. Copyright Office. Unauthorized copying of this file, via any medium
 * is strictly prohibited.
 */
/* eslint-disable react-hooks/exhaustive-deps */
import React, { useEffect, useState } from 'react';
import useCounter from './hooks/useCounter';

const Counter = () => {
  const [count, setCount] = useState<number>(0);
  const { counter, loading, getCounter } = useCounter();

  useEffect(() => {
    getCounter();
  }, []);

  useEffect(() => {
    setCount(counter);
  }, [counter]);

  if (loading) {
    return <div>Loading...</div>;
  }

  return (
    <>
      <button type="button" data-testid="buttonQa" onClick={() => setCount(count + 1)}>
        Button
      </button>
      <div data-testid="counter">Count is: {count}</div>
    </>
  );
};

export default Counter;
