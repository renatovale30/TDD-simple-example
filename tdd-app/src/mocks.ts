import { rest } from "msw";
import pokemons from "./fixtures/pokemon.json";

export const getCount = rest.get("*/getCount", (req, res, ctx) => {
  return res(ctx.json("100"));
});

export const getPokemons = rest.get("*/getPokemons", (req, res, ctx) => {
  return res(ctx.json(pokemons));
});
