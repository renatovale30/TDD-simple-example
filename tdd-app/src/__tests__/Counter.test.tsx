import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import Counter from '../Counter';
import {setupServer} from 'msw/node'
import { getCount } from '../mocks';

const renderComponent = () => {
    return render(<Counter />)
}

describe('Counter suite', () => {
    beforeAll(() => {
        const server = setupServer(getCount);
        server.listen();
    });

    test('Should increment count when button is clicked', async () => {
         renderComponent();

         const button = await screen.findByTestId("buttonQa");

         fireEvent.click(button);

         expect(screen.getByTestId('counter').textContent).toEqual("Count is: 101");
    });

    test("Should render correctly", async() => {
        const {container} = renderComponent();

        await screen.findByTestId("buttonQa");

        expect(container).toMatchSnapshot();
    })
})