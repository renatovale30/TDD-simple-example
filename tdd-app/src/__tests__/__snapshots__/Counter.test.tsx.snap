// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter suite Should render correctly 1`] = `
<div>
  <button
    data-testid="buttonQa"
    type="button"
  >
    Button
  </button>
  <div
    data-testid="counter"
  >
    Count is: 
    100
  </div>
</div>
`;
